### PR TITLE
Add script & Dockerfile for building image from local source code

### PR DESCRIPTION
**Pull request by bszaf**
Goal of PR is not to replace existing solution, but to provide script, that allows to run `script.sh <path_to_mim>` and just wait for image.

However it may be used in future to build solutions replacing `builder` and `member` scripts.

### CODE REVIEW

File: multistage_build.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+
+MONGOOSEIM_DOCKER_DIR="./"
```
Comment (mkacper) [inline multistage_build.sh]: It's correct only if you start the script from it's directory.

File: multistage_build.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+# We assume MongooseIM directory is git repository. Get current ref and version
+GIT_REF=`git rev-parse --short HEAD`
+VERSION=$(tools/generate_vsn.sh)
+
+# If image tag is not provided with env variable, template it out
+# branch and git ref
+if [ -z "$IMAGE_TAG" ]; then
+    GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD)
+    IMAGE_TAG="mongooseim/mongooseim:${GIT_BRANCH}-${GIT_REF}"
```
Comment (mkacper) [inline multistage_build.sh]: You can check if a branch name contains any "\\", if so replace or remove them cause presence of such a sign can cause docker crash.

Comment (mkacper) [reply multistage_build.sh]: OR just inform a user that using "/" is forbidden :)

File: README.md [hunk 3]
```diff
@@ -201,3 +201,12 @@ It is possible to pass custom image tag with environmental variable:
 IMAGE_TAG=my_mongooseim_docker_build \
   ./multistage_build.sh <path_to_moongooseim_root_dir>
 ```
+
+#### Pitfalls
+
+By default `multistage_build.sh` script uses git branch and git reference
+as a image tag, e.g.: `new_feature-c36dce4fd`. Branch names may contain
```
Comment (mkacper) [inline README.md]: "an image"

File: README.md [hunk 4]
```diff
@@ -201,3 +201,12 @@ It is possible to pass custom image tag with environmental variable:
...
+characters, which are invalid tags. For instance `/` cannot be present
+in tag name. In such case, it custom tag has to be explicitly set
```
Comment (mkacper) [inline README.md]: "in a tag name", "in such a case its"

File: multistage_build.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+MONGOOSEIM_DOCKER_DIR=$(dirname $(realpath ${0}))
```
Comment (kanes115) [inline multistage_build.sh]: `realpath` will not be available on mac os, will it, @bszaf?

Comment (kanes115) [reply multistage_build.sh]: ok, it is in coreutils, it's ok

**Comment by kanes115** [comment]
@bszaf as far as I see, there's no way to include pem files right now. Would we want to make it possible?

**Comment by chrzaszcz** [comment]
Closing because it is outdated.

**Pull request closed without merging**